### PR TITLE
Fix log and error messages for get identity feature

### PR DESCRIPTION
**Pull request by tzheleznyak**
Fix log messages by @shulifink comments

CC: @hilagross

### CODE REVIEW

**Review by hilagross — CHANGES_REQUESTED** [review]
@tzheleznyak 
I was going over the PR and I was missing many fixes that Shuli asked for

General comments that she gave and I think we should take into consideration are:

- JWT JWKS and URI are always capitalized (unless, I guess, they are used as a parameter or value that is defined with lowercase letters)
- Successfully did this..., Failed to do that... (Rather than "Did this successfully") - for consistency

The PR I didn't find the changes are:
https://github.com/cyberark/conjur/pull/2136
https://github.com/cyberark/conjur/pull/2155
https://github.com/cyberark/conjur/pull/2157 - some comments are in and some not
https://github.com/cyberark/conjur/pull/2141
https://github.com/cyberark/conjur/pull/2139 - some comments wasn't addressed

please go over Shuli's comments (just search for shulifink and you'll find them easily)

File: app/domain/logs.rb [hunk 1]
```diff
@@ -351,42 +351,42 @@ module AuthnJwt
...
 
       CheckingJwtClaimToValidate = ::Util::TrackableLogMessageClass.new(
-        msg: "Checking if JWT claim '{0-claim-name}' is mandatory to validate",
+        msg: "Checking if JWT claimת '{0-claim-name}' is mandatory to validate...",
         code: "CONJ00070D"
       )
 
       FetchingJwtConfigurationValue = ::Util::TrackableLogMessageClass.new(
-        msg: "Fetching jwt signing key uri from '{0-resource-id}'",
+        msg: "Fetching JWT signing key URI from authenticator configuration'{0-resource-id}'...",
```
Comment (hilagross) [inline app/domain/logs.rb:364]: @shulifink  please review

File: app/domain/logs.rb [hunk 2]
```diff
@@ -351,42 +351,42 @@ module AuthnJwt
       )
 
       AddingJwtClaimToValidate = ::Util::TrackableLogMessageClass.new(
-        msg: "Adding JWT claim '{0-claim-name}' to list of mandatory claims to be validated",
+        msg: "Adding JWT claim, '{0-claim-name}' to list of mandatory claims to be validated...",
```
Comment (hilagross) [inline app/domain/logs.rb:354]: missing ,
Adding JWT claim, '{0-claim-name}'**,** to list of mandatory claims to be validated..."

File: app/domain/logs.rb [hunk 3]
```diff
@@ -351,42 +351,42 @@ module AuthnJwt
...
         code: "CONJ00069D"
       )
 
       CheckingJwtClaimToValidate = ::Util::TrackableLogMessageClass.new(
-        msg: "Checking if JWT claim '{0-claim-name}' is mandatory to validate",
+        msg: "Checking if JWT claimת '{0-claim-name}' is mandatory to validate...",
```
Comment (hilagross) [inline app/domain/logs.rb:359]: you're in Hebrew :) and notice there are 2 ,
Checking if JWT claim, '{0-claim-name}', must be validated...

**Review by hilagross — CHANGES_REQUESTED** [review]
Also in error.rb - InvalidIssuerConfiguration - please change Issuer to "issuer"

File: app/domain/logs.rb [hunk 4]
```diff
@@ -351,42 +351,42 @@ module AuthnJwt
...
 
       FetchJwtUriKeysSuccess = ::Util::TrackableLogMessageClass.new(
-        msg: "Fetched JWKS successfully",
+        msg: "Successfully fetched JWKS",
         code: "CONJ00073D"
       )
 
       ValidatingJwtSigningKeyConfiguration = ::Util::TrackableLogMessageClass.new(
-        msg: "Validating Jwt signing key uri configuration",
+        msg: "Validating JWT signing key URI configuration...",
```
Comment (shulifink) [inline app/domain/logs.rb:379]: don;t understand what this. means? what is a signing key URI?

File: app/domain/logs.rb [hunk 5]
```diff
@@ -351,42 +351,42 @@ module AuthnJwt
...
         code: "CONJ00074D"
       )
 
       FetchingProviderUriSigningKey = ::Util::TrackableLogMessageClass.new(
-        msg: "Fetching provider-uri signing key",
+        msg: "Fetching PROVIDER-URI signing key...",
```
[the remark this answers is not in the document]
Comment (shulifink) [reply app/domain/logs.rb:384]: i agree. but i still need to understand what this signing key is

File: app/domain/logs.rb [hunk 6]
```diff
@@ -351,42 +351,42 @@ module AuthnJwt
...
         code: "CONJ00075D"
       )
 
       FetchingJwksUriSigningKey = ::Util::TrackableLogMessageClass.new(
-        msg: "Fetching jwks-uri signing key",
+        msg: "Fetching JWKS-URI signing key...",
```
[the remark this answers is not in the document]
Comment (shulifink) [reply app/domain/logs.rb:389]: agree

File: app/domain/logs.rb [hunk 7]
```diff
@@ -405,7 +405,7 @@ module AuthnJwt
       )
 
       SigningKeysFetchedFromCache = ::Util::TrackableLogMessageClass.new(
-        msg: "Fetched signing keys from cache successfully",
+        msg: "Successfully fetched signing keys from cache",
```
Comment (shulifink) [inline app/domain/logs.rb:408]: fine

File: app/domain/logs.rb [hunk 8]
```diff
@@ -415,7 +415,7 @@ module AuthnJwt
       )
 
       ValidatedToken = ::Util::TrackableLogMessageClass.new(
-        msg: "Validated token successfully",
+        msg: "Successfully validated token",
```
Comment (shulifink) [inline app/domain/logs.rb:418]: Baruch HaShem

**Review by shulifink — CHANGES_REQUESTED** [review]

File: app/domain/logs.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (tzheleznyak) [reply app/domain/logs.rb:384]: @hilagross you are right

File: app/domain/errors.rb [hunk 9]
```diff
@@ -420,8 +416,7 @@ module AuthnJwt
       )
 
       InvalidUriConfiguration = ::Util::TrackableErrorClass.new(
-        msg: "Signing key uri authenticator configuration is invalid. It should configured as authenticator variables: " \
-              "one of the following: '{0-resource-name}','{1-resource-name}'",
+        msg: "Signing key URI authenticator configuration is invalid",
         code: "CONJ00178E"
```
[the remark this answers is not in the document]
Comment (tzheleznyak) [reply app/domain/errors.rb:420]: @Avi-Eyal  took this troubleshooting
https://github.com/cyberark/conjur/pull/2157#discussion_r636023271

File: app/domain/logs.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (tzheleznyak) [reply app/domain/logs.rb:384]: provider-uri = OIDC provider hostname
Note: Fetch key from provider uri base on OIDC protocol

File: app/domain/logs.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (tzheleznyak) [reply app/domain/logs.rb:389]: Done

File: app/domain/logs.rb (hunk 1 above)
Comment (tzheleznyak) [reply app/domain/logs.rb:364]: This log indicating if we fetching singing key from provider uri or from jwks. This is why the parameter for

File: app/domain/logs.rb (hunk 2 above)
Comment (tzheleznyak) [reply app/domain/logs.rb:354]: Done. Strange its not pushed

File: app/domain/logs.rb (hunk 1 above)
Comment (tzheleznyak) [reply app/domain/logs.rb:364]: Fetching JWT signing key URI from authenticator configuration conjur/authn-jwt/some-vendor/jwks-uri

File: app/domain/logs.rb (hunk 3 above)
Comment (tzheleznyak) [reply app/domain/logs.rb:359]: Goot catch thanks

File: app/domain/errors.rb [hunk 10]
```diff
@@ -420,8 +416,7 @@ module AuthnJwt
       )
 
       InvalidUriConfiguration = ::Util::TrackableErrorClass.new(
-        msg: "Signing key uri authenticator configuration is invalid. It should configured as authenticator variables: " \
-              "one of the following: '{0-resource-name}','{1-resource-name}'",
+        msg: "Signing key URI authenticator configuration is invalid",
```
[the remark this answers is not in the document]
Comment (tzheleznyak) [reply app/domain/errors.rb:419]: This means one of this two cases happened:
Both JWKS URI configured and Provider URI configured
Or neither of them configured.

This both cases are a misconfiguration of our authenticator

[the remark this answers is not in the document]
Comment (tzheleznyak) [reply app/domain/errors.rb:419]: I will add this to troubleshooting

[the remark this answers is not in the document]
Comment (tzheleznyak) [reply app/domain/errors.rb:419]: image

File: app/domain/logs.rb (hunk 4 above)
Comment (tzheleznyak) [reply app/domain/logs.rb:379]: Validating the one of this parameters and just one of this parameters is configured in the policy of the authenticator
`jwks-uri `and `provider-uri`

File: app/domain/logs.rb [hunk 11]
```diff
@@ -306,17 +306,17 @@ module AuthnJwt
...
 
       CHECKING_IDENTITY_FIELD_EXISTS = ::Util::TrackableLogMessageClass.new(
-        msg: "Checking if the field exists",
+        msg: "Checking if the field exists...",
         code: "CONJ00061D"
       )
 
       CREATING_AUTHENTICATION_PARAMETERS_OBJECT = ::Util::TrackableLogMessageClass.new(
-        msg: "Creating authentication parameters objects",
+        msg: "Creating authentication parameters objects...",
```
[the remark this answers is not in the document]
Comment (hilagross) [reply app/domain/logs.rb:319]: @tzheleznyak

**Review by hilagross — CHANGES_REQUESTED** [review]
@tzheleznyak  according to Shuli's comments in [here](https://github.com/cyberark/conjur/pull/2162)
CONJ00188E - should be "HTTP response **format** is invalid"

**Review by hilagross — APPROVED** [review]

File: app/domain/errors.rb (hunk 10 above)
[the remark this answers is not in the document]
Comment (shulifink) [reply app/domain/errors.rb:424]: @tzheleznyak  - let's talk about this.

File: app/domain/logs.rb [hunk 12]
```diff
@@ -266,37 +266,37 @@ module AuthnAzure
...
 
       RetrievedIssuerValue = ::Util::TrackableLogMessageClass.new(
-        msg: "Retrieved issuer with value '{0}'",
+        msg: "Retrieved \"issuer\" with value '{0}'",
         code: "CONJ00055D"
       )
 
       ParsingIssuerFromUri = ::Util::TrackableLogMessageClass.new(
-        msg: "Parsing issuer value from URI '{0}'",
+        msg: "Parsing \"issuer\" value from URI '{0}'...",
```
[the remark this answers is not in the document]
Comment (shulifink) [reply app/domain/logs.rb:289]: @tzheleznyak  - waiting to see an example

File: app/domain/logs.rb (hunk 2 above)
Comment (shulifink) [reply app/domain/logs.rb:354]: @tzheleznyak  can you provide example?

File: app/domain/logs.rb (hunk 1 above)
Comment (shulifink) [reply app/domain/logs.rb:364]: fine.

File: app/domain/logs.rb (hunk 4 above)
Comment (shulifink) [reply app/domain/logs.rb:379]: @tzheleznyak  - Let's discuss....

File: app/domain/logs.rb [hunk 13]
```diff
@@ -351,42 +351,42 @@ module AuthnJwt
...
 
       ValidatingJwtSigningKeyConfiguration = ::Util::TrackableLogMessageClass.new(
-        msg: "Validating Jwt signing key uri configuration",
+        msg: "Validating JWT signing key URI configuration...",
         code: "CONJ00074D"
       )
 
       FetchingProviderUriSigningKey = ::Util::TrackableLogMessageClass.new(
-        msg: "Fetching provider-uri signing key",
+        msg: "Fetching provider-uri signing key...",
```
Comment (shulifink) [inline app/domain/logs.rb:384]: fine

File: app/domain/logs.rb [hunk 14]
```diff
@@ -351,42 +351,42 @@ module AuthnJwt
...
         code: "CONJ00075D"
       )
 
       FetchingJwksUriSigningKey = ::Util::TrackableLogMessageClass.new(
-        msg: "Fetching jwks-uri signing key",
+        msg: "Fetching jwks-uri signing key...",
```
Comment (shulifink) [inline app/domain/logs.rb:389]: fine

**Review by shulifink — CHANGES_REQUESTED** [review]

File: app/domain/errors.rb (hunk 10 above)
[the remark this answers is not in the document]
Comment (shulifink) [reply app/domain/errors.rb:424]: "Signing key URI configuration is invalid"

(as discussed with @tzheleznyak )

File: app/domain/logs.rb (hunk 12 above)
[the remark this answers is not in the document]
Comment (shulifink) [reply app/domain/logs.rb:289]:
```suggestion
        msg: "Parsing \"issuer\" value from {0}...",
```

File: app/domain/logs.rb (hunk 2 above)
Comment (shulifink) [inline app/domain/logs.rb:354]:
```suggestion
        msg: "Adding JWT claim, '{0-claim-name}', to list of mandatory claims to be validated...",
```

File: app/domain/logs.rb (hunk 4 above)
Comment (shulifink) [reply app/domain/logs.rb:379]:
```suggestion
        msg: "Validating signing key URI configuration...",
```

File: app/domain/logs.rb [hunk 15]
```diff
@@ -351,42 +351,42 @@ module AuthnJwt
...
 
       CheckingJwtClaimToValidate = ::Util::TrackableLogMessageClass.new(
-        msg: "Checking if JWT claim '{0-claim-name}' is mandatory to validate",
+        msg: "Checking if JWT claim '{0-claim-name}' is mandatory to validate...",
         code: "CONJ00070D"
       )
 
       FetchingJwtConfigurationValue = ::Util::TrackableLogMessageClass.new(
-        msg: "Fetching jwt signing key uri from '{0-resource-id}'",
+        msg: "Fetching JWT signing key URI from authenticator configuration'{0-resource-id}'...",
```
Comment (shulifink) [inline app/domain/logs.rb:364]:
```suggestion
        msg: "Fetching '{0-resource-id}' variable value from configuration...",
```

**Review by shulifink — CHANGES_REQUESTED** [review]
As per our discussion, the team need to decide where things are being taken from. 
If the configuration/policy/role are all different things, then fine, but from our discussion they are all referring to the same thing, so the same term should be used.

Users will always regard different terms as exactly that, different.

File: app/domain/logs.rb (hunk 2 above)
Comment (tzheleznyak) [reply app/domain/logs.rb:354]: Done

**Pull request merged**